### ISSUE TITLE
Async test failing locally with `connection refused`

### ISSUE THREAD

**Issue by adigitoleo** [JuliaIO/JSON.jl#324]
For me (Arch Linux, julia 1.6.1) the async test is failing with `LoadError: IOError: connect: connection refused (ECONNREFUSED)`. I think [this stackoverflow answer](https://stackoverflow.com/a/60328592) might be relevant, but I'm not very experienced with async stuff.

Just tested on Win10 and got the same error.

My julia version (Arch Linux):

```
julia> versioninfo()
Julia Version 1.6.1
Commit 6aaedecc44* (2021-04-23 05:59 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8705G CPU @ 3.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.0 (ORCJIT, skylake)

```

Julia version (Win10):

```
julia> versioninfo()
Julia Version 1.6.1
Commit 6aaedecc44 (2021-04-23 05:59 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.01 (ORCJIT, skylake)
```

<details><summary>Test log:</summary>

```
(@v1.6) pkg> test JSON
     Testing JSON
      Status `/tmp/jl_3iTj73/Project.toml`
  [864edb3b] DataStructures v0.18.9
  [53c48c17] FixedPointNumbers v0.8.4
  [682c06a0] JSON v0.21.1
  [6fe1bfb0] OffsetArrays v1.10.0
  [69de0a69] Parsers v1.1.0
  [ade2ca70] Dates `@stdlib/Dates`
  [8ba89e20] Distributed `@stdlib/Distributed`
  [a63ad114] Mmap `@stdlib/Mmap`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [8dfed614] Test `@stdlib/Test`
  [4ec0a83e] Unicode `@stdlib/Unicode`
      Status `/tmp/jl_3iTj73/Manifest.toml`
  [79e6a3ab] Adapt v3.3.1
  [34da2185] Compat v3.30.0
  [864edb3b] DataStructures v0.18.9
  [53c48c17] FixedPointNumbers v0.8.4
  [682c06a0] JSON v0.21.1
  [6fe1bfb0] OffsetArrays v1.10.0
  [bac558e1] OrderedCollections v1.4.1
  [69de0a69] Parsers v1.1.0
  [0dad84c5] ArgTools `@stdlib/ArgTools`
  [56f22d72] Artifacts `@stdlib/Artifacts`
  [2a0f44e3] Base64 `@stdlib/Base64`
  [ade2ca70] Dates `@stdlib/Dates`
  [8bb1440f] DelimitedFiles `@stdlib/DelimitedFiles`
  [8ba89e20] Distributed `@stdlib/Distributed`
  [f43a241f] Downloads `@stdlib/Downloads`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [b27032c2] LibCURL `@stdlib/LibCURL`
  [76f85450] LibGit2 `@stdlib/LibGit2`
  [8f399da3] Libdl `@stdlib/Libdl`
  [37e2e46d] LinearAlgebra `@stdlib/LinearAlgebra`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [a63ad114] Mmap `@stdlib/Mmap`
  [ca575930] NetworkOptions `@stdlib/NetworkOptions`
  [44cfe95a] Pkg `@stdlib/Pkg`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [ea8e919c] SHA `@stdlib/SHA`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [1a1011a3] SharedArrays `@stdlib/SharedArrays`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [2f01184e] SparseArrays `@stdlib/SparseArrays`
  [10745b16] Statistics `@stdlib/Statistics`
  [fa267f1f] TOML `@stdlib/TOML`
  [a4e569a6] Tar `@stdlib/Tar`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
  [4ec0a83e] Unicode `@stdlib/Unicode`
  [deac9b47] LibCURL_jll `@stdlib/LibCURL_jll`
  [29816b5a] LibSSH2_jll `@stdlib/LibSSH2_jll`
  [c8ffd9c3] MbedTLS_jll `@stdlib/MbedTLS_jll`
  [14a3606d] MozillaCACerts_jll `@stdlib/MozillaCACerts_jll`
  [83775a58] Zlib_jll `@stdlib/Zlib_jll`
  [8e850ede] nghttp2_jll `@stdlib/nghttp2_jll`
  [3f19e933] p7zip_jll `@stdlib/p7zip_jll`
     Testing Running tests...
Test Summary: | Pass  Total
Parser        |   86     86
Test Summary: | Pass  Total
Serializer    |   49     49
async: Error During Test at /home/admin/.julia/packages/JSON/3rsiS/test/runtests.jl:69
  Got exception outside of a @test
  LoadError: IOError: connect: connection refused (ECONNREFUSED)
  Stacktrace:
    [1] wait_connected(x::TCPSocket)
      @ Sockets /build/julia/src/julia-1.6.1/usr/share/julia/stdlib/v1.6/Sockets/src/Sockets.jl:532
    [2] connect(::TCPSocket, ::String, ::Int64)
      @ Sockets /build/julia/src/julia-1.6.1/usr/share/julia/stdlib/v1.6/Sockets/src/Sockets.jl:567
    [3] connect(host::String, port::Int64)
      @ Sockets /build/julia/src/julia-1.6.1/usr/share/julia/stdlib/v1.6/Sockets/src/Sockets.jl:552
    [4] top-level scope
      @ ~/.julia/packages/JSON/3rsiS/test/async.jl:36
    [5] include(fname::String)
      @ Base.MainInclude ./client.jl:444
    [6] macro expansion
      @ ~/.julia/packages/JSON/3rsiS/test/runtests.jl:70 [inlined]
    [7] macro expansion
      @ /build/julia/src/julia-1.6.1/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
    [8] macro expansion
      @ ~/.julia/packages/JSON/3rsiS/test/runtests.jl:70 [inlined]
    [9] macro expansion
      @ /build/julia/src/julia-1.6.1/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [10] top-level scope
      @ ~/.julia/packages/JSON/3rsiS/test/runtests.jl:64
   [11] include(fname::String)
      @ Base.MainInclude ./client.jl:444
   [12] top-level scope
      @ none:6
  in expression starting at /home/admin/.julia/packages/JSON/3rsiS/test/async.jl:36
Test Summary:  | Pass  Error  Total
Integration    |   71      1     72
  async        |           1      1
  indentation  |    2             2
  JSON Checker |   68            68
ERROR: LoadError: Some tests did not pass: 71 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/admin/.julia/packages/JSON/3rsiS/test/runtests.jl:62
ERROR: Package JSON errored during testing
```
</details>

**Comment by adigitoleo** [comment]
Whoops, looks like this has been fixed in 12209499b08eb126a045b9c39a280c9ebef2bc14

**Comment by adigitoleo** [comment]
Tests are passing using latest master. I'll close this. Maybe you can tag a new release soon?
